### ISSUE TITLE
[Accessibility]: "Add saved search" button on empty saved searches panel doesn't activate with spacebar

### ISSUE THREAD

**Issue by limitedmage** [sourcegraph/sourcegraph#34721]
### Audit type

Keyboard navigation

### User journey audit issue

https://github.com/sourcegraph/sourcegraph/issues/https://github.com/sourcegraph/sourcegraph/issues/34189

### Problem description

1. Open Sourcegraph Server (not .com)
2. If the saved searches panel is not empty, delete any saved searches (both own and org)
3. Focus on the "+ Create a saved search" button
4. Press space, observe nothing happens

![image](https://user-images.githubusercontent.com/206864/166071773-7f833ee8-8269-4f4d-8854-4adf08887b36.png)

### Expected behavior

The button should activate

### Additional details

The button is actually a link, so enter needs to be pressed instead. Either the button should be a real button, or the link should react to space

### Assigning labels

- [X] Please give this issue an estimate by applying a label like `estimate/Xd`, where X is the estimated number of days it will take to complete.
- [X] If this issue is specific to a specific Sourcegraph product, please assign the appropriate team label to this issue.
- [ ] If this issue will require input from designers in order to complete, please assign the label `needs-design`.
- [X] If you are confident that this issue should be fixed by GitStart, please assign the label `gitstart`.

### Owner

This issue will be fixed by GitStart. I have assigned the label 'gitstart' and added the assignee 'gitstart-sourcegraph'.

**Comment by sourcegraph-bot-2** [comment]
Heads up @benvenker @lguychard - the "team/search-product" label was applied to this issue.
